### ISSUE TITLE
Websocket server disconnects right after connect

### ISSUE THREAD

**Issue by avently** [chronoxor/NetCoreServer#62]
Hello, Ivan. Trying to make wss + https server working but got a strange situation. I cloned the repo and ran WssChatServer project. What I got is disconnections right after connect. And more importantly is that behaviour is not constant. So after some tries I can connect and receive back messages that I sent to the chat but in the other try the chat disconnects me. I made a screenrecord just to show what I mean.
Also I added a `Console.WriteLine(e)` in your code with exception that makes these disconnections happen.

Video inside the zip archive since GitHub doesn't allow video files to be posted without archive.

[NetCoreServer.zip](https://github.com/chronoxor/NetCoreServer/files/4641177/NetCoreServer.zip)

OS: Linux, test project is the same, I just changed path to certificate to absolute path because without it the app couldn't find the cert and printed SSL errors.

Disconnects happen here:
https://github.com/chronoxor/NetCoreServer/blob/master/source/NetCoreServer/SslSession.cs#L585

```
System.NotSupportedException:  The ReadAsync method cannot be called when another read operation is pending.
   at System.Net.Security.SslStream.ReadAsyncInternal[TReadAdapter](TReadAdapter adapter, Memory`1 buffer)
   at System.Threading.Tasks.TaskToApm.End[TResult](IAsyncResult asyncResult)
   at System.Net.Security.SslStream.EndRead(IAsyncResult asyncResult)
   at NetCoreServer.SslSession.ProcessReceive(IAsyncResult result)
```

What do you think, есть мысли, как можно исправить ситуацию? А то я не силен в C# пока что.

**Comment by dmrayt** [comment]
I have the same problem with SslChatClient / SslChatServer.

The exception occurs in SslChatClient but I was able to resolve it by removing the line ``send(message);`` in ``SslChatServer.ChatSession.OnHandshaked()``.

**Comment by avently** [comment]
@dmrayt the sample I mentioned doesn't have an override of onhandshaked and doesn't have a send(message) inside

**Comment by dmrayt** [comment]
In my case the exception doesn't occurs anymore with version 3.0.14.

**Comment by chronoxor** [comment]
Should be fixed in 3.0.14

**Comment by avently** [comment]
@chronoxor works perfectly. thank you!
